hcl2template/types.source.go

```diff
@@ -57,12 +58,23 @@ func (cfg *PackerConfig) startBuilder(source *SourceBlock, ectx *hcl.EvalContext
 		return nil, diags, nil
 	}
 
-	generatedVars, warning, err := builder.Prepare(source.builderVariables(), decoded)
+	// Note: HCL prepares inside of the Start func, but Json does not. Json
+	// builds are instead prepared only in command/build.go
+	// TODO: either make json prepare when plugins are loaded, or make HCL
+	// prepare at a later step, to make builds from different template types
+	// easier to reason about.
```

Comment:
I think this comment should be in here:
https://github.com/hashicorp/packer/blob/52b2151b21b362833257df7e04f8c6ad50501628/packer/run_interfaces.go#L11-L16

Reply:
We could mention it there too, but I think a person trying to read and understand this code would be looking here and now in the interfaces file, so it's useful to explain here too. (at least until now -- I like your idea of instead moving the json configs to match the HCL configs. I'll make a follow-up PR to do that, and then this comment can be deleted.